### PR TITLE
Meal 32 Update user model and write User API

### PR DESCRIPTION
**Pull request by seanyles**

### CODE REVIEW

File: src/main/java/com/app/eece4792mealmaster/controllers/RecipeController.java [hunk 1]
```diff
@@ -12,13 +12,13 @@
 
 // TODO: Add EC2 hostname to allowed origins
 @RestController
-@CrossOrigin(origins = {"http://localhost:3000"}, allowCredentials = "true")
+@CrossOrigin(origins = {"*","http://localhost:3000"}, allowCredentials = "true")
```
Comment (otuffaha) [inline src/main/java/com/app/eece4792mealmaster/controllers/RecipeController.java]: just wondering what difference this makes

File: src/main/java/com/app/eece4792mealmaster/repositories/UserRepository.java [hunk 2]
```diff
@@ -19,4 +21,7 @@ public User findUserByCredentials(@Param("usernameEmail") String usernameEmail,
 
   @Query("SELECT user FROM User user WHERE user.email=:email")
   public User findUserByEmail(@Param("email") String email);
+
+  @Query("SELECT user from User user WHERE user.username LIKE :searchTerms%")
```
Comment (otuffaha) [inline src/main/java/com/app/eece4792mealmaster/repositories/UserRepository.java]: is the % intentional? is this like a regex key for queries?

**Comment by otuffaha** [comment]
ok with this being merged. hard to give a comprehensive review without being able to have the ability to test on front end.

File: src/main/java/com/app/eece4792mealmaster/repositories/UserRepository.java (hunk 2 above)
Comment (seanyles) [reply src/main/java/com/app/eece4792mealmaster/repositories/UserRepository.java]: % means that if I type a it will query anything that starts with a

**Pull request merged**